core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -32,7 +32,14 @@ private[v1] class ApplicationListResource(uiRoot: UIRoot) {
       @DefaultValue("3000-01-01") @QueryParam("maxDate") maxDate: SimpleDateParam,
       @QueryParam("limit") limit: Integer)
   : Iterator[ApplicationInfo] = {
-    val allApps = uiRoot.getApplicationInfoList
+    val allApps = {
```

Comment:
view is good point. 
However, it doesn't apply here because uiRoot.getApplicationInfoList.view is too late and the transformation of the whole list within getApplicationInfoList() still happens before the .view method. Changing getApplicationInfoList to return a view is not a good idea since it breaks its original intention. 

To make it one line, I can change the method signature to getApplicationInfoList(limit: java.lang.Integer) which is a little bit ugly.